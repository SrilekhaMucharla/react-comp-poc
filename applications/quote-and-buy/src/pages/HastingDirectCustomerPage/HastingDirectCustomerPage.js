import React, { Component } from 'react';
// import { connect } from 'react-redux';
import PropTypes from 'prop-types';
// import './HastingDirectCustomerPage.scss';
import {
    HDInfoCard, HDLabel
} from 'hastings-components';
import { Col, Row } from 'react-bootstrap';
import { AnalyticsHDButton as HDButton, AnalyticsHDOverlayPopup as HDOverlayPopup } from '../../web-analytics';
import BackNavigation from '../Controls/BackNavigation/BackNavigation';
import tipCirclePurple from '../../assets/images/icons/tip_circle_purple.svg';
// import { sendPageData } from '../../redux-thunk/actions';
import * as Message from './HastingDirectCustomerPage.messages';

class HastingDirectCustomerPage extends Component {
    static propTypes = {
        history: PropTypes.shape({
            push: PropTypes.func
        }).isRequired,
        pageMetadata: PropTypes.shape({
            page_name: PropTypes.string.isRequired,
            page_type: PropTypes.string.isRequired,
            sales_journey_type: PropTypes.string.isRequired
        }).isRequired,
    };

    // pageID = 'HastingDirectCustomerPage';

    // componentDidMount() {
    //     // eslint-disable-next-line react/prop-types
    //     const { dispatch } = this.props;
    //     const dataObj = { page_name: this.pageID };
    //     dispatch(sendPageData(dataObj));
    // }

    handleWizardFlow = () => {
        const { history } = this.props;
        history.push({
            pathname: '/intro-car-page',
            state: { singleCar: true }
        });
    };

    tooltipOverlay = (id) => (
        <HDOverlayPopup
            // eslint-disable-next-line react/destructuring-assignment
            webAnalyticsView={{ ...this.props.pageMetadata, page_section: `${Message.acceptancecriteriaprefix} Info` }}
            webAnalyticsEvent={{ event_action: `${Message.acceptancecriteriaprefix} Info` }}
            id={id}
            overlayButtonIcon={(
                <>
                    <span className="acceptance-criteria-prefix">{Message.acceptancecriteriaprefix}</span>
                    <p className="acceptance-criteria-link">
                        {Message.acceptancecriteria}
                    </p>
                </>
            )}
        >
            <div id="introOverlayContainer" className="introcustomeroverlaycontainer">
                <p id="introOverlayHeaderParagraph" className="introcustomeroverlayheader">
                    {Message.overlayHeader}
                </p>
                <p>
                    {Message.overlayBodyTitle1}
                </p>
                <ul>
                    {[
                        Message.overlayBodyLiseElementa,
                        Message.overlayBodyLiseElementb,
                        Message.overlayBodyLiseElementc,
                        Message.overlayBodyLiseElementd,
                        Message.overlayBodyLiseElemente,
                        Message.overlayBodyLiseElementf,
                        Message.overlayBodyLiseElementg,
                        Message.overlayBodyLiseElementh,
                    // eslint-disable-next-line react/no-array-index-key
                    ].map((el, index) => (<li key={index}><span>{el}</span></li>))}
                </ul>
                <p>
                    {Message.overlayBodyTitle2}
                </p>
                <ol>
                    {[
                        Message.overlayBodyListElementPart2a,
                        Message.overlayBodyListElementPart2b,
                        Message.overlayBodyListElementPart2c,
                    ].map((el) => (<li><span>{el}</span></li>))}
                </ol>
                <p>
                    {Message.overlayBodyTitle3}
                </p>
            </div>
        </HDOverlayPopup>
    );

    render() {
        return (
            <div className="introcustomerpagecontainer" id="introCustomerPageContainer">
                <div className="introcomponents" id="introComponents">
                    <Row className="grid-container" id="introGridContainer">
                        <Col lg={6} md={6} xs={12} className="item-left" id="introItemLeft">
                            <div className="item-left-container" id="introItemLeftContainer">
                                <BackNavigation />
                                <HDLabel text={Message.hello} size="xl" />
                                <div className="customer-button first-btn">
                                    <HDButton
                                        webAnalyticsEvent={{ event_action: Message.hello }}
                                        id="existing-customer-button"
                                        variant={Message.cardbutton}
                                        label={Message.oldcustomer}
                                        // eslint-disable-next-line react/jsx-no-bind
                                        onClick={this.handleWizardFlow.bind(this)} />
                                </div>
                                <div className="customer-button">
                                    <HDButton
                                        webAnalyticsEvent={{ event_action: Message.hello }}
                                        id="new-customer-button"
                                        variant={Message.cardbutton}
                                        label={Message.newcustomer}
                                        disabled />
                                </div>
                                <div className="tip-main-left" id="introTipMainLeft">
                                    <HDInfoCard image={tipCirclePurple} paragraphs={[Message.alreadycarbike]} />
                                </div>
                                <div className="acceptance-criteria" id="introAcceptanceCriteria">
                                    <HDLabel text="" size="sm" icon={this.tooltipOverlay('introAcceptanceCriteriaLink')} iconPosition="l" />
                                </div>
                            </div>
                        </Col>
                        <Col lg={6} md={6} xs={12} className="item-right" id="introItemRight">
                            <div className="item-right-container" id="introItemRightContainer">
                                <div className="tip-main-right" id="introTipMainRight">
                                    <HDInfoCard
                                        isRight
                                        title={Message.titleright}
                                        paragraphs={[Message.tosaveyourself, Message.tosaveyourselfparagraph]}
                                        notes={Message.rightnotes} />
                                </div>
                            </div>
                        </Col>
                    </Row>
                </div>
            </div>
        );
    }
}

HastingDirectCustomerPage.propTypes = {};

export default HastingDirectCustomerPage;
