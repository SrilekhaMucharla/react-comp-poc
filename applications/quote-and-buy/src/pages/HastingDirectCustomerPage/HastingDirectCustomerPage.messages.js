/* eslint-disable max-len */
export const hello = 'Hi, do you already have car insurance with us?';
export const cardbutton = 'cardbutton';
export const newcustomer = 'Yes, I have a policy';
export const oldcustomer = "No, I'm new here";
export const titleleft = 'Tip!';
export const alreadycarbike = 'If you already have a car policy with us (even if it\'s not started yet), we can use some of your details to speed up your quote.';
export const existingPolicy = 'If you already have a live or pending car policy, we can use your existing driver details to save you time getting a quote.';
export const acceptancecriteriaprefix = 'To get a quote today, you need to meet ';
export const acceptancecriteria = 'certain criteria';
export const titleright = 'Save time and money...';
export const tosaveyourself = 'If you\'ve got your driving licence handy, you\'ll be able to scan it and get a faster quote. We can also calculate any discount you qualify for.';
export const tosaveyourselfparagraph = 'Most customers save around Â£68!*';
export const rightnotes = '*Based on online quoted premiums from July to September 2020.';
export const overlayHeader = 'Car insurance acceptance criteria';
export const overlayBodyTitle1 = 'To buy this policy today, you must meet these requirements:';
export const overlayBodyLiseElementa = 'All drivers must permanently live in England, Scotland, Wales or Northern Ireland. We\'re sorry, we can\'t offer a quote for anyone who lives in the Isle of Man or the Channel Islands.';
export const overlayBodyLiseElementb = 'All cars must be registered in the UK.';
export const overlayBodyLiseElementc = 'The DVLA/DVANI have been advised of any notifiable conditions (for all drivers) and have agreed to the issue of a licence.';
export const overlayBodyLiseElementd = 'No driver has any unspent non-motoring convictions.';
export const overlayBodyLiseElemente = 'No driver will use any vehicle for business purposes that relate to a driving occupation, such as being a courier, delivering fast food or working in the motor trade (or any other trade that involves hire and reward).';
export const overlayBodyLiseElementf = 'Between all drivers, there are no more than five accidents, claims or losses in the last five years on any vehicle.';
export const overlayBodyLiseElementg = 'Between all drivers there are no more than five motoring convictions or endorsements (including fixed penalties) in the last five years. This includes any prosecutions that are pending.';
export const overlayBodyLiseElementh = 'No car is currently impounded by the police or any other authority.';
export const overlayBodyTitle2 = 'You can only claim no claims discount if:';
export const overlayBodyListElementPart2a = 'It\'s been earned in your name (as a policyholder) on a private car policy.';
export const overlayBodyListElementPart2b = 'It\'s been earned in the UK in the last two years.';
export const overlayBodyListElementPart2c = 'It\'s not already being used.';
export const overlayBodyTitle3 = 'The rules are slightly different for anyone in the armed forces who have been on service abroad recently. If this applies to you, please call us on 0333 321 9765 for help.';
