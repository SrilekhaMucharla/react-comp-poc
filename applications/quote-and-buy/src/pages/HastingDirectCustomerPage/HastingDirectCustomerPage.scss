.introcustomerpagecontainer {
  width: 100%;;
  margin-top: $header-height;
  background: $app-theme;
  .row{
    margin-right: 0;
    margin-left: 0;
  }

  @include media-breakpoint-between(xs, sm) {
    margin-top: $header-height-mobile;
  }

  .info-intro-customer-page {
    display: flex;
    & > p {
      margin-right: 4.2%;
    }
  }

  .go-back {
    pointer-events: none;
    opacity: 0.5;
  }

  .introcustomeroverlaycontainer {
    padding: 10px 30px;
    color: $info-color;
    font-family: $font-face-generic-regular;

    .introcustomeroverlayheader {
      border: 0;
      color: $app-theme;
      font-style: normal;
      font-weight: bold;
      font-size: 1.6rem;
      line-height: 1.2;
      color: $app-theme;
      margin-bottom: 2rem;
      font-family: $font-face-generic-regular;
    }
    ul,
    ol {
      li {
        font-family: $font-face-generic-regular;
        padding: 2%;
      }
      span {
        vertical-align: initial;
      }
    }
    ol {
      list-style: none;
      counter-reset: li;
      li {
        counter-increment: li;
      }
      li::before {
        content: counter(li);
        color: $link-color;
        display: inline-block;
        width: 1em;
        margin-left: -1em;
      }
    }
    ul {
      padding-left: 5.5%;
      @include media-breakpoint-between(xs, sm) {
        padding-left: 7.5%;
      }
      li {
        list-style: none;
        list-style-position: inside;
        position: relative;
        span {
          vertical-align: initial;
        }
        &::before {
          font-weight: bold;
          display: inline-block;
          width: 1em;
          margin-left: -1em;
          color: $link-color;
          font-size: 1.1em;
        }
      }
    }

    .introcustomeroverlaybody {
      font-style: normal;
      font-weight: normal;
      font-size: $font-size-md;
      line-height: 1.5;
      color: $blue-light;
    }
  }
}

.grid-container {
  display: flex;
  margin: 0;

  @include media-breakpoint-down(lg) {
    display: block;
  }
}

.item-left,
.item-right {
  
  display: block;
  text-align: center;

  @include media-breakpoint-down(lg) {
    min-width: 100%;
    max-width: 100%;
    width: 100%;
    display: inline-block; 
  }
}

.item-left {
  background: $app-theme;
  //height: 90vh;
  width:50%;
  @include media-breakpoint-between(xs, sm) {
    //height: 100%;
  }
  span.label-text {
    font-size: 2.625rem;
    line-height: 1.2;
    @include media-breakpoint-down(lg) {
      font-size: 2.1rem;
      line-height: 1.2;
    }
  }

  .item-left-container {
    // padding: 10% 20% 25% 25%;
    width: 87.5%;
    cursor: pointer;
    text-align: center;
    display: inline-block;
    vertical-align: middle;
    //padding-top: 12%;

    @include media-breakpoint-between(xs, sm) {
      width: 90%;
    }

    .disable {
      color: rgba(0, 0, 0, 0.4);
    }

    .sub-header {
      font-size: $font-size-xxl;
      font-weight: bold;
      margin-bottom: 38px;
      text-align: left;
    }
    .hd-overlay-btn {
      text-decoration: none;
    }
    
    .customer-button {     
      @include media-breakpoint-up(md) {
        display: inline-block;
        margin-bottom: 0;
        width: 48.6%;
        &.first-btn {
          margin-right: 2.8%;
        }
      }

      margin-bottom: 14px;
      .btn-cardbutton {
        font-family: $font-face-generic-regular !important;
        font-weight: 600;
        @include media-breakpoint-up(md){
          width: 100%;
        }
      }

      span {
        padding-left: 31px;
      }

      @include media-breakpoint-down(lg) {        
        margin-bottom: 27px;
      }
    }

    .tip-main-left {
      margin-top: 30px;
      width: 100%;;

      @include media-breakpoint-down(lg) {    
        margin-top: 20px;
      }

      div {
        display: flex;
        background-color: $blue-dark-4;
        color: $white-color;
        img {
          display: inline-flex;
          align-self: baseline;
        }
        p {
          display: inline-flex;
          padding-left: 10px;
          font-size: $font-size-md;
          line-height: 1.5;
          text-align: left;
          font-family: $font-face-generic-regular;
        }
      }
    }

    .acceptance-criteria {
      margin-top: 14px;
      span:first-child {
        @include media-breakpoint-between(xs, sm) {
          margin-bottom: 0;
        }
      }
      .label-text {
        color: $white-color;
        font-size: $font-size-md;
        line-height: 1.5;
        font-weight: bold;
        text-align: left;
        p.acceptance-criteria-link {
          text-decoration: underline;
          text-decoration-color: $link-color;
        }
        span.acceptance-criteria-prefix {
          text-decoration: none !important;
        }
      }

      @include media-breakpoint-down(lg) {     
          margin-top: 20px;
      }
    }
  }
}

.item-right {
  background-color: transparent;
  z-index: 1;
  @include media-breakpoint-down(lg) { 
    padding-bottom: 10%;
    margin-top: 0;
    width:50%;
    position:absolute;
  }

  .item-right-container {
    cursor: pointer;
    text-align: center;
    display: inline-block;
    vertical-align: middle;
    width: 86%;
    padding-top: 20%;

    @include media-breakpoint-between(xs, sm) {
      width: 90%;
    }

    @include media-breakpoint-down(lg) {   
      padding: 0%;
    }
  }

  .tip-main-right {
    border-radius: 8px;
    text-align: left;
    div {
      background-color: $white-color;
    }

    .right {
      padding: 28px;
    }

    .info-card-title {
      font-style: normal;
      font-size: 1.625rem;
      line-height: 1.2;
      color: $app-theme;
      margin: 0;
      font-family: $font-face-generic-regular;
      font-weight: bold;
    }

    .info-card-paragraph {
      font-family: $font-face-generic-regular;
      font-size: 1rem;
      line-height: 1.5;
      color: $app-theme;
      margin-top: 2%;
    }

    .info-card-notes {
      font-family: $font-face-generic-regular;
      font-style: normal;
      font-weight: normal;
      font-size: .75rem;
      line-height: 1.25;
      color: $blue-light;
      margin-top: 10%;
    }

    // will be removed in future commit
    .tip-main-header {
      font-size: 24px;
      color: $gray-dark-3;
      font-weight: 700;
      font-style: normal;
      line-height: 30px;
    }

    // will be removed in future commit
    .tip-main-body {
      padding-top: 28px;
      font-size: $font-size-regular;
      color: $gray-dark-3;
      font-weight: normal;
      font-style: normal;
      line-height: 16px;
    }

    // will be removed in future commit
    .tip-main-footer {
      padding-top: 10px;
      font-size: $font-size-sm;
      color: $gray-light-18;
      font-weight: normal;
      font-style: normal;
      line-height: 17px;
    }
  }
}
