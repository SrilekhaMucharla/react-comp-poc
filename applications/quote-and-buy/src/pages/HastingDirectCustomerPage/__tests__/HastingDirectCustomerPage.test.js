import React from 'react';
import { mount, shallow } from 'enzyme';
import HastingDirectCustomerPage from '../HastingDirectCustomerPage';
import createPortalRoot from '../../wizard-pages/__helpers__/test/createPortalRoot';


describe('<HastingDirectCustomerPage />', () => {
    createPortalRoot();

    test('render component', () => {
        const wrapper = shallow(
            <HastingDirectCustomerPage />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('button at index 0 is not disabled', () => {
        const wrapper = mount(
            <HastingDirectCustomerPage />
        );
        expect(wrapper.find('#existing-customer-button').at(0).prop('disabled')).toBeFalsy();
    });

    test('button at index 1 is disabled', () => {
        const wrapper = mount(
            <HastingDirectCustomerPage />
        );
        expect(wrapper.find('#new-customer-button').at(0).prop('disabled')).toBeTruthy();
    });

    test('it displays tooltipOverlay on click', async () => {
        const wrapper = mount(
            <HastingDirectCustomerPage />
        );
        const acceptanceLink = wrapper.find('.hd-overlay-btn');

        await act(async () => {
            acceptanceLink
                .simulate('click');
        });
        wrapper.update();
        expect(wrapper.find('.overlay').exists()).toBe(true);
    });

    test('on handleWizardFlow click', async () => {
        const wrapper = mount(
            <HastingDirectCustomerPage />
        );
        const mockRedirect = jest.fn();
        wrapper.instance().handleWizardFlow = mockRedirect;
        wrapper.update();
        wrapper.instance().handleWizardFlow();
        expect(mockRedirect).toHaveBeenCalled();
    });
});
