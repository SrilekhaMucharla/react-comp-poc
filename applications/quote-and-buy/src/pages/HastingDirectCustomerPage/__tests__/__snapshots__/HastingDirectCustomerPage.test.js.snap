// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HastingDirectCustomerPage /> render component 1`] = `
<div
  className="introcustomerpagecontainer"
  id="introCustomerPageContainer"
>
  <div
    className="introcomponents"
    id="introComponents"
  >
    <Row
      className="grid-container"
      id="introGridContainer"
      noGutters={false}
    >
      <Col
        className="item-left"
        id="introItemLeft"
        lg={6}
        md={6}
        xs={12}
      >
        <div
          className="item-left-container"
          id="introItemLeftContainer"
        >
          <BackNavigation
            className=""
            onClick={null}
            onKeyPress={null}
          />
          <HDLabel
            additionalText={null}
            icon={null}
            iconPosition={null}
            size="xl"
            text="Hi, do you already have car insurance with us?"
            theme="dark"
          />
          <div
            className="customer-button first-btn"
          >
            <ForwardRef
              id="existing-customer-button"
              label="No, I'm new here"
              onBeforeOpen={null}
              onBlur={null}
              onCancel={null}
              onChange={null}
              onClick={[Function]}
              onConfirm={null}
              onDelete={null}
              onEdit={null}
              onFocus={null}
              onSelect={null}
              variant="cardbutton"
              webAnalyticsEvent={
                Object {
                  "event_action": "Hi, do you already have car insurance with us?",
                }
              }
            />
          </div>
          <div
            className="customer-button"
          >
            <ForwardRef
              disabled={true}
              id="new-customer-button"
              label="Yes, I have a policy"
              onBeforeOpen={null}
              onBlur={null}
              onCancel={null}
              onChange={null}
              onClick={null}
              onConfirm={null}
              onDelete={null}
              onEdit={null}
              onFocus={null}
              onSelect={null}
              variant="cardbutton"
              webAnalyticsEvent={
                Object {
                  "event_action": "Hi, do you already have car insurance with us?",
                }
              }
            />
          </div>
          <div
            className="tip-main-left"
            id="introTipMainLeft"
          >
            <HDInfoCard
              className={null}
              image="tip_circle_purple.svg"
              isRight={false}
              notes={null}
              paragraphs={
                Array [
                  "If you already have a car policy with us (even if it's not started yet), we can use some of your details to speed up your quote.",
                ]
              }
              title={null}
            />
          </div>
          <div
            className="acceptance-criteria"
            id="introAcceptanceCriteria"
          >
            <HDLabel
              additionalText={null}
              icon={
                <ForwardRef
                  id="introAcceptanceCriteriaLink"
                  onBeforeOpen={null}
                  overlayButtonIcon={
                    <React.Fragment>
                      <span
                        className="acceptance-criteria-prefix"
                      >
                        To get a quote today, you need to meet 
                      </span>
                      <p
                        className="acceptance-criteria-link"
                      >
                        certain criteria
                      </p>
                    </React.Fragment>
                  }
                  webAnalyticsEvent={
                    Object {
                      "event_action": "To get a quote today, you need to meet  Info",
                    }
                  }
                  webAnalyticsView={
                    Object {
                      "page_section": "To get a quote today, you need to meet  Info",
                    }
                  }
                >
                  <div
                    className="introcustomeroverlaycontainer"
                    id="introOverlayContainer"
                  >
                    <p
                      className="introcustomeroverlayheader"
                      id="introOverlayHeaderParagraph"
                    >
                      Car insurance acceptance criteria
                    </p>
                    <p>
                      To buy this policy today, you must meet these requirements:
                    </p>
                    <ul>
                      <li>
                        <span>
                          All drivers must permanently live in England, Scotland, Wales or Northern Ireland. We're sorry, we can't offer a quote for anyone who lives in the Isle of Man or the Channel Islands.
                        </span>
                      </li>
                      <li>
                        <span>
                          All cars must be registered in the UK.
                        </span>
                      </li>
                      <li>
                        <span>
                          The DVLA/DVANI have been advised of any notifiable conditions (for all drivers) and have agreed to the issue of a licence.
                        </span>
                      </li>
                      <li>
                        <span>
                          No driver has any unspent non-motoring convictions.
                        </span>
                      </li>
                      <li>
                        <span>
                          No driver will use any vehicle for business purposes that relate to a driving occupation, such as being a courier, delivering fast food or working in the motor trade (or any other trade that involves hire and reward).
                        </span>
                      </li>
                      <li>
                        <span>
                          Between all drivers, there are no more than five accidents, claims or losses in the last five years on any vehicle.
                        </span>
                      </li>
                      <li>
                        <span>
                          Between all drivers there are no more than five motoring convictions or endorsements (including fixed penalties) in the last five years. This includes any prosecutions that are pending.
                        </span>
                      </li>
                      <li>
                        <span>
                          No car is currently impounded by the police or any other authority.
                        </span>
                      </li>
                    </ul>
                    <p>
                      You can only claim no claims discount if:
                    </p>
                    <ol>
                      <li>
                        <span>
                          It's been earned in your name (as a policyholder) on a private car policy.
                        </span>
                      </li>
                      <li>
                        <span>
                          It's been earned in the UK in the last two years.
                        </span>
                      </li>
                      <li>
                        <span>
                          It's not already being used.
                        </span>
                      </li>
                    </ol>
                    <p>
                      The rules are slightly different for anyone in the armed forces who have been on service abroad recently. If this applies to you, please call us on 0333 321 9765 for help.
                    </p>
                  </div>
                </ForwardRef>
              }
              iconPosition="l"
              size="sm"
              text=""
              theme="dark"
            />
          </div>
        </div>
      </Col>
      <Col
        className="item-right"
        id="introItemRight"
        lg={6}
        md={6}
        xs={12}
      >
        <div
          className="item-right-container"
          id="introItemRightContainer"
        >
          <div
            className="tip-main-right"
            id="introTipMainRight"
          >
            <HDInfoCard
              className={null}
              image={null}
              isRight={true}
              notes="*Based on online quoted premiums from July to September 2020."
              paragraphs={
                Array [
                  "If you've got your driving licence handy, you'll be able to scan it and get a faster quote. We can also calculate any discount you qualify for.",
                  "Most customers save around Â£68!*",
                ]
              }
              title="Save time and money..."
            />
          </div>
        </div>
      </Col>
    </Row>
  </div>
</div>
`;
